File: e-credit-backend/src/main/java/org/example/ecreditbackend/model/Piece.java
package org.example.ecreditbackend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name="piece")
@Data
@NoArgsConstructor
public class Piece {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_piece")

    private Integer idPiece;
   @OneToOne(cascade = CascadeType.ALL)
    private Demande demande;

    private String doc1;
    private Boolean statut1;
    private Boolean obligatoire1;

    private String doc2;
    private Boolean statut2;
    private Boolean obligatoire2;



}

File: e-credit-frontend/src/app/Models/Piece.ts
import { Demande } from "./Demande";

 
  export class Piece {
    idPiece!: number;
    demande!: Demande;
  
    doc1: string;
    statut1: boolean;
    obligatoire1: boolean;
  
    doc2: string;
    statut2: boolean;
    obligatoire2: boolean;
  
    constructor()
    {
 
      this.doc1 = "document1";
      this.statut1 = false;
      this.obligatoire1 = false;
      
      this.doc2 = "document2";
      this.statut2 = false;
      this.obligatoire2 = false;
    }
  }
  
 
  

